TradeItIosTicketSDK2/TradeItLinkedBrokerCache.swift

```diff
@@ -1,62 +1,51 @@
-
-    private let ACCOUNTS_KEY = "ACCOUNTS"
-    private let ACCOUNTS_LAST_UPDATED_KEY = "ACCOUNTS_LAST_UPDATED"
-    private let ACCOUNT_NAME_KEY = "ACCOUNT_NAME"
-    private let ACCOUNT_NUMBER_KEY = "ACCOUNT_NUMBER"
-    private let ACCOUNT_INDEX_KEY = "ACCOUNT_INDEX"
-    private let ACCOUNT_BASE_CURRENCY_KEY = "ACCOUNT_BASE_CURRENCY"
-    private let ACCOUNT_ENABLED_KEY = "ACCOUNT_ENABLED"
-    private let ACCOUNTS_LINK_DELAY_ERROR_KEY = "ACCOUNTS_LINK_DELAY_ERROR"
-    private let BALANCE_LAST_UPDATED_KEY = "BALANCE_LAST_UPDATED"
-    private let BALANCE_BUYING_POWER_KEY = "BALANCE_BUYING_POWER"
-    private let FX_BALANCE_BUYING_POWER_KEY = "FX_BALANCE_BUYING_POWER"
-
-    private let ACCOUNT_ENABLED_VALUE = "ENABLED"
-    private let ACCOUNT_DISABLED_VALUE = "DISABLED"
-
-    internal static var _userDefaults = UserDefaults(suiteName: "it.trade")!
-    internal var userDefaults: UserDefaults {
-        get {
-            return TradeItLinkedBrokerCache._userDefaults
-        }
-    }
 
+    private let LINKED_BROKER_CACHE_KEY = "TRADE_IT_LINKED_BROKER_CACHE_"
+  
     func cache(linkedBroker: TradeItLinkedBroker?) {
         guard let linkedBroker = linkedBroker else { return }
-
-        var linkedBrokerCache = userDefaults.dictionary(forKey: LINKED_BROKER_CACHE_KEY) as? SerializedLinkedBrokers ?? SerializedLinkedBrokers()
-
-        let serializedLinkedBroker = serialize(linkedBroker: linkedBroker)
-
-        linkedBrokerCache[linkedBroker.linkedLogin.userId] = serializedLinkedBroker
-
-        self.userDefaults.set(linkedBrokerCache, forKey: LINKED_BROKER_CACHE_KEY)
+        let key = LINKED_BROKER_CACHE_KEY+linkedBroker.linkedLogin.userId
+        
+        let cachedLinkedBroker = CachedLinkedBroker(linkedBroker: linkedBroker)
+        TradeItKeychain.save(cachedLinkedBroker.toJSONString(), forKey: key)
+        
+        removeOldCache(linkedBroker: linkedBroker)
     }
 
     func syncFromCache(linkedBroker: TradeItLinkedBroker) {
-        guard let linkedBrokerCache = self.userDefaults.dictionary(forKey: LINKED_BROKER_CACHE_KEY) as? SerializedLinkedBrokers
-            , let serializedLinkedBroker = linkedBrokerCache[linkedBroker.linkedLogin.userId] as SerializedLinkedBroker?
-            else { return }
-
-        if let serializedAccounts = serializedLinkedBroker[ACCOUNTS_KEY] as? [SerializedLinkedBrokerAccount] {
-            let accounts = deserialize(
-                serializedAccounts: serializedAccounts,
-                forLinkedBroker: linkedBroker
+        let key = LINKED_BROKER_CACHE_KEY+linkedBroker.linkedLogin.userId
+        guard let json = TradeItKeychain.getStringForKey(key)
+             ,let cachedLinkedBroker = TradeItResultTransformer.transform(targetClassType: CachedLinkedBroker.self, json: json)
+            else { return}
+        
+        linkedBroker.accounts = cachedLinkedBroker.accounts.map { cachedAccount in
+            let account = TradeItLinkedBrokerAccount(
```

Comment:
Can we use JSONModel to deserialize the JSON directly into the nested account/balance objects instead of creating them mapping all the fields over manually?